Sources/SwiftProtobuf/FieldNameMap.swift

```diff
@@ -0,0 +1,105 @@
+// ProtobufRuntime/Sources/Protobuf/FieldNameMap.swift - Bidirectional field number/name mapping
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+
+/// An immutable bidirectional mapping between field names and numbers, used
+/// for various text-based serialization (JSON and text).
+public struct FieldNameMap: ExpressibleByDictionaryLiteral {
+
+  /// A "bundle" of names for a particular field. We use different "packed"
+  /// representations to minimize the amount of string data that we store in the
+  /// binary.
+  ///
+  /// TODO: Remove the Swift field name since it's only used for debugging
+  /// purposes and use the proto name instead at those usage sites.
+  public enum Names {
+
+    /// The proto name and the JSON name are the same string
+    case same(proto: String, swift: String)
```

Comment:
Scanning the generated code, most of the time, all three of these seem to be identical.  Should there be a case for that as well?